[Bootcamp] Set default value of TCPStore world_size to None in pybind definition

kryanchun opened the pull request:
- Set the default value of TCPStore optional world_size arg to None
- Updated the corresponding test case

Fixes #74488

H-Huang commented on test/distributed/test_store.py:
```diff
@@ -259,15 +258,18 @@ def _multi_worker_helper(self, world_size):
-        for i in range(world_size):
-            self._create_client(i, addr, port, world_size)
+
+        if world_size and world_size > 0:
```
nit: this could be simplified by keeping for loop and doing something like:

```python
num_indices = world_size if world_size else 1
for i in range(num_indices):
   self._create_client(i, addr, port, world_size)
```

H-Huang reviewed (commented):
Overall looks good, test failures look unrelated.

H-Huang commented on test/distributed/test_store.py:
```diff
@@ -259,15 +258,18 @@ def _multi_worker_helper(self, world_size):
         port = server_store.port
-        world_size = random.randint(5, 10) if world_size == -1 else world_size
-        for i in range(world_size):
-            self._create_client(i, addr, port, world_size)
+
+        if world_size and world_size > 0:
+            for i in range(world_size):
+                self._create_client(i, addr, port, world_size)
+        else:
+            self._create_client(0, addr, port, None)
```
nit: this could be simplified by keeping for loop and doing something like:

```python
num_indices = world_size if world_size else 1
for i in range(num_indices):
   self._create_client(i, addr, port, world_size)
```

H-Huang reviewed (approved):
LGTM! Thanks for adding these changes

kryanchun committed:
[Bootcamp] Set default value of TCPStore world_size to None in pybind definition (#77277)

Summary:
- Set the default value of TCPStore optional world_size arg to None
- Updated the corresponding test case

Fixes https://github.com/pytorch/pytorch/issues/74488

Pull Request resolved: https://github.com/pytorch/pytorch/pull/77277

Reviewed By: H-Huang

Differential Revision: D36347267

Pulled By: kryanchun

fbshipit-source-id: f9aa147428b07d25b8dc9084c0d3261b0dfd8bdd

H-Huang commented:
@pytorchbot merge this on green
